### ISSUE TITLE
AboutModal should trap focus

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1389]
Currently the AboutModal does not trap focus like the regular Modal does. The same method used for trapping focus in the Modal should be used for AboutModal.

Details about how to trap focus are documented in the [PF accessibility guide](https://pf4.patternfly.org/accessibility-guide#trapping-focus).

In addition to what's documented there, when a modal is dismissed by the user, focus should be placed on the button that opened the modal.

**Comment by jgiardino** [comment]
I was reviewing the Modal with @jenny-s51 and noticed an issue with how Modal traps focus for screen readers. 

I think the PR where we implemented this was https://github.com/patternfly/patternfly-react/pull/1011. I'm not sure if there were other updates made since then that intended to modify this behavior, or if updates since then were made unintentionally, but the behavior required to trap focus for screen reader users is no longer working.

The method we use for trapping focus is documented in the [accessibility guide](https://pf4.patternfly.org/accessibility-guide#trapping-focus):
> ... apply `aria-hidden="true"` to the parent wrapping element of the page contents. Note that the modal element of the component must not be a descendent of this element with `aria-hidden="true"` and should be included as a sibling to this element.

Currently, if I open a modal, I see that `aria-hidden` is applied to every child of the `<body>` element on the page, including the `<div>` that has the modal. This breaks screen reader accessibility.

This issue with modal should be fixed before updating AboutModal to use a similar implementation.

**Comment by jenny-s51** [comment]
please see #2428

**Comment by tlabaj** [comment]
closed by #2405
